### DISCUSSION TITLE
undefined is not a constructor (evaluating 'new WebSocket(this._endPointURL())') - Safari

### DISCUSSION THREAD

**Discussion by pulse-ventures** [supabase/supabase#28036] [Questions]
This error seems to pop up when people are using the app. 

this is the relevant error in chrome `WebSocket is not a constructor`

```
.../@supabase/realtime-js/dist/module/RealtimeClient.js at line 122

if (NATIVE_WEBSOCKET_AVAILABLE) {
    this.conn = new WebSocket(this._endPointURL());
    this.setupConnection();
    return;
}
```

**Comment by pulse-ventures** [comment]
it may be linked to Service workers or deployments of new code etc

**Comment by GaryAustin1** [comment]
No one seems to have info for you on this and you are not presenting much info for someone to help.  If you are having an issue with a running app, you probably need to present more context on what you are doing, framework, languages, etc.  If you think there is a bug in realtime then an issue in the realtime-js or realtime repository may be appropriate, but you will need a lot more info for someone to help.

Otherwise if no one hits your exact same error message don't expect much traction.

**Comment by pulse-ventures** [comment]
* Using NextJS and RX DB with supabase replication. 

Everything works most of the time and even when this error happens i think everything fixes itself and the application continues to run. Its just that this error is constant at some point in the lifecycle
